modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/info/item/provider/CommerceOrderItemInfoItemFieldValuesProvider.java

```diff
@@ -0,0 +1,284 @@
+import com.liferay.info.field.InfoFieldValue;
+import com.liferay.info.item.InfoItemFieldValues;
+import com.liferay.info.item.InfoItemReference;
+import com.liferay.info.item.field.reader.InfoItemFieldReaderFieldSetProvider;
+import com.liferay.info.item.provider.InfoItemFieldValuesProvider;
+import com.liferay.info.localized.InfoLocalizedValue;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceContextThreadLocal;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.LocaleUtil;
+import com.liferay.template.info.item.provider.TemplateInfoItemFieldSetProvider;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alessio Antonio Rendina
+ */
+@Component(service = InfoItemFieldValuesProvider.class)
+public class CommerceOrderItemInfoItemFieldValuesProvider
+	implements InfoItemFieldValuesProvider<CommerceOrderItem> {
+
+	@Override
+	public InfoItemFieldValues getInfoItemFieldValues(
+		CommerceOrderItem commerceOrderItem) {
+
+		return InfoItemFieldValues.builder(
+		).infoFieldValues(
+			_getCommerceOrderItemInfoFieldValues(commerceOrderItem)
+		).infoFieldValues(
+			_expandoInfoItemFieldSetProvider.getInfoFieldValues(
+				CommerceOrderItem.class.getName(), commerceOrderItem)
+		).infoFieldValues(
+			_templateInfoItemFieldSetProvider.getInfoFieldValues(
+				CommerceOrderItem.class.getName(), commerceOrderItem)
+		).infoFieldValues(
+			_infoItemFieldReaderFieldSetProvider.getInfoFieldValues(
+				CommerceOrderItem.class.getName(), commerceOrderItem)
+		).infoItemReference(
+			new InfoItemReference(
+				CommerceOrderItem.class.getName(),
+				commerceOrderItem.getCommerceOrderItemId())
+		).build();
+	}
+
+	private List<InfoFieldValue<Object>> _getCommerceOrderItemInfoFieldValues(
+		CommerceOrderItem commerceOrderItem) {
+
+		ThemeDisplay themeDisplay = _getThemeDisplay();
+
+		Locale locale = themeDisplay.getLocale();
+
+		List<InfoFieldValue<Object>> commerceOrderItemInfoFieldValues =
+			new ArrayList<>();
+
+		commerceOrderItemInfoFieldValues.add(
+			new InfoFieldValue<>(
+				CommerceOrderItemInfoItemFields.companyIdInfoField,
+				commerceOrderItem.getCompanyId()));
+		commerceOrderItemInfoFieldValues.add(
+			new InfoFieldValue<>(
+				CommerceOrderItemInfoItemFields.createDateInfoField,
+				commerceOrderItem.getCreateDate()));
+		commerceOrderItemInfoFieldValues.add(
+			new InfoFieldValue<>(
+				CommerceOrderItemInfoItemFields.decimalQuantityInfoField,
+				String.valueOf(commerceOrderItem.getDecimalQuantity())));
```

Comment:
same here
https://liferay.atlassian.net/browse/COMMERCE-11743

Reply:
String.valueOf() method calls the toString() method of the object. So, it should work when the decimal quantity will be a BigDecimal